sakai-evaluation-tool/src/java/org/sakaiproject/evaluation/tool/reporting/XLSReportExporter.java

```diff
@@ -79,223 +95,627 @@ public void setMessageLocator(MessageLocator locator) {
-     * (non-Javadoc)
+    /**
+     * Build the .xls report in the new (section based) format.
      * 
-     * @see
-     * org.sakaiproject.evaluation.tool.reporting.ReportExporter#buildReport(org.sakaiproject.evaluation
-     * .model.EvalEvaluation, java.lang.String[], java.lang.String, java.io.OutputStream)
+     * @param evaluation
+     * @param groupIDs
+     * @param outputStream
      */
-    public void buildReport(EvalEvaluation evaluation, String[] groupIds, String evaluateeId, OutputStream outputStream) {
-    	
-        /*
-         * Logic for creating this view 1) make tidl 2) get DTIs for this eval from tidl 3) use DTIs
-         * to make the headers 4) get responseIds from tidl 5) loop over response ids 6) loop over
-         * DTIs 7) check answersmap for an answer, if there put in cell, if missing, insert blank 8)
-         * done
-         */
-
-    	
-    	//Make sure responseAggregator is using this messageLocator
-        responseAggregator.setMessageLocator(messageLocator);
-
-         Boolean instructorViewAllResults = (boolean) evaluation.getInstructorViewAllResults();
-         String currentUserId = commonLogic.getCurrentUserId();
-         String evalOwner = evaluation.getOwner();
-
-	boolean isCurrentUserAdmin = commonLogic.isUserAdmin(currentUserId);
-
-        // 1 Make TIDL
-    	TemplateItemDataList tidl = getEvalTIDL(evaluation, groupIds);
-    	// 2: get DTIs for this eval from tidl
-        List<DataTemplateItem> dtiList = tidl.getFlatListOfDataTemplateItems(true);
- 
-        Workbook wb = new XSSFWorkbook();
+    private void buildReportSectionAware( EvalEvaluation evaluation, String[] groupIDs, OutputStream outputStream )
+    {
+        // Get permission to view, current user and eval owner
+        Boolean instructorViewAllResults = evaluation.getInstructorViewAllResults();
+        String currentUserId = commonLogic.getCurrentUserId();
+        String evalOwner = evaluation.getOwner();
+
+        // Allow columns greater than 255 (EVALSYS-775)
+        TemplateItemDataList tidl = getEvalTIDL( evaluation, groupIDs );
+        List<DataTemplateItem> dtiList = tidl.getFlatListOfDataTemplateItems( true );
+        Workbook wb;
+        if( dtiList.size() < 256 )
+        {
+            wb = new HSSFWorkbook();
+        }
+        else
+        {
+            wb = new XSSFWorkbook();
+        }
         creationHelper = wb.getCreationHelper();
-        
-        Sheet sheet = wb.createSheet(messageLocator.getMessage("reporting.xls.sheetname"));
 
-        // Title Style
+        // Title style
+        Sheet courseSheet = wb.createSheet( messageLocator.getMessage( "viewreport.xls.courseSheet.name" ) );
+        Sheet instructorSheet = wb.createSheet( messageLocator.getMessage( "viewreport.xls.instructorSheet.name" ) );
         Font font = wb.createFont();
-        font.setFontHeightInPoints((short) 12);
-        font.setBoldweight(Font.BOLDWEIGHT_BOLD);
+        font.setFontHeightInPoints( (short) 12 );
```

Comment:
wondering if we need all these short casts, perhaps kill two birds with one stone and use meaningful constants? short PT12 = (short) 12;
and something for all the createCell ( (short) 0 ) that happens later on